modules/ppcp-order-tracking/services.php

```diff
@@ -100,12 +100,13 @@
 		}
 	},
 	'order-tracking.is-module-enabled'         => static function ( ContainerInterface $container ): bool {
-		$order_id = isset( $_GET['post'] ) ? (int) $_GET['post'] : ''; // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		$order_id = wc_clean( wp_unslash( $_GET['id'] ?? $_GET['post'] ?? '' ) ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+
 		if ( empty( $order_id ) ) {
 			return false;
 		}
 
-		$meta = get_post_meta( $order_id, PayPalGateway::ORDER_ID_META_KEY, true );
+		$meta = get_post_meta( (int) $order_id, PayPalGateway::ORDER_ID_META_KEY, true );
```

Comment:
We should not use WP post meta here but order api to get order meta, HPOS uses new order tables instead of post tables:
```
		$order = wc_get_order( $post_id );
		if ( ! is_a( $order, WC_Order::class ) ) {
			return false;
		}

       $order->get_meta(...
```

Reply:
Thanks, refactored 4747d9d2243b4c9f78b45478a1657d716279ce87 and ee51504f5f479d97c5c69c21733c3614960f522d